### ISSUE TITLE
systemd --test crashes when output is closed early

### ISSUE THREAD

**Issue by keszybz** [systemd/systemd#6383]
Filing this not to forget:
```           PID: 16536 (systemd)
           UID: 1000 (zbyszek)
           GID: 1000 (zbyszek)
        Signal: 6 (ABRT)
     Timestamp: Sun 2017-07-16 19:34:29 EDT (17s ago)
  Command Line: /usr/lib/systemd/systemd --test
    Executable: /usr/lib/systemd/systemd
 Control Group: /user.slice/user-1000.slice/session-3.scope
          Unit: session-3.scope
         Slice: user-1000.slice
       Session: 3
     Owner UID: 1000 (zbyszek)
       Boot ID: b0c3e9794ac04ef69401fb4378d12e6d
    Machine ID: ad18f69b80264b52bb3b766240742383
      Hostname: fedora
       Storage: /var/lib/systemd/coredump/core.systemd.1000.b0c3e9794ac04ef69401fb4378d12e6d.16536.1500248069000000.lz4
       Message: Process 16536 (systemd) of user 1000 dumped core.
                
                Stack trace of thread 16536:
                #0  0x00007fc87a4cf66b raise (libc.so.6)
                #1  0x00007fc87a4d1470 abort (libc.so.6)
                #2  0x000055e146fffd3c log_assert_failed_realm (systemd)
                #3  0x000055e146fee1fd safe_close (systemd)
                #4  0x000055e146fee2a7 safe_close_pair (systemd)
                #5  0x000055e146ee3952 manager_close_idle_pipe (systemd)
                #6  0x000055e146ee67a2 manager_free (systemd)
                #7  0x000055e146ee183f main (systemd)
                #8  0x00007fc87a4b94da __libc_start_main (libc.so.6)
                #9  0x000055e146ed96ea _start (systemd)
```

**Comment by poettering** [comment]
what do you mean by "output is closed early" specifically?

That the pipe fd connected to stdout is close()d while the test is still running?

**Comment by evverx** [comment]
The crash can be reproduced by running `SYSTEMD_LESS=+q systemd --test`. It seems to be a regression because `v231` works well without crashing.

**Comment by poettering** [comment]
FIx waiting in #6810.
